pkg/ds/farm.go

```diff
@@ -306,9 +312,12 @@ func (dsf *Farm) handleDSChanges(changes dsstore.WatchedDaemonSets, quitCh <-cha
 					dsf.logger.Errorf("Updated daemon set '%s' contends with %s", dsFields.ID, dsIDContended)
 					newDS, err := dsf.dsStore.Disable(dsFields.ID)
 					if err != nil {
-						dsf.logger.Errorf("Error occurred when trying to delete daemon set: %v", err)
+						dsf.logger.Errorf("Error occurred when trying to disable daemon set: %v", err)
 						continue
 					}
+					if dsFields.ID != newDS.ID {
```

Comment:
have you seen this happen? seems like a weird thing to detect and `dsStore.Disable()` should guarantee that this can't happen